bin/wasm-node/rust/src/ffi.rs

```diff
@@ -512,29 +512,53 @@ fn init(
     ));
 }
 
+#[derive(Debug)]
+pub(crate) struct Request {
+    pub(crate) method_name: Box<[u8]>,
+    pub(crate) source_id: u32,
+}
+
 lazy_static::lazy_static! {
-    static ref JSON_RPC_CHANNEL: (mpsc::UnboundedSender<Box<[u8]>>, futures::lock::Mutex<mpsc::UnboundedReceiver<Box<[u8]>>>) = {
+    static ref JSON_RPC_CHANNEL: (mpsc::UnboundedSender<Request>, futures::lock::Mutex<mpsc::UnboundedReceiver<Request>>) = {
+        let (tx, rx) = mpsc::unbounded();
+        (tx, futures::lock::Mutex::new(rx))
+    };
+    static ref UNSUBSCRIBE_ALL_CHANNEL: (mpsc::UnboundedSender<u32>, futures::lock::Mutex<mpsc::UnboundedReceiver<u32>>) = {
```

Comment:
Please use only one channel and turn `Request` into an enum. We don't want to lose the ordering of the things sent by the outside.